odoo/orm/domains.py

```diff
@@ -820,7 +820,7 @@ def _optimize(self, model: BaseModel, full: bool) -> Domain:
             return DomainCondition(parent_fname, 'any', parent_domain)
 
         # handle non-stored fields (replace by searchable/stored items)
-        if not field.store:
+        if not field.sqlable:
```

Comment:
#204106 also changes the way this works to use search on all fields. It should be an independent change from this one and could be merged first to avoid changing this line at all.